src/polynomial.h

```diff
@@ -111,6 +112,12 @@ Polynomial<T>::Polynomial(gf::Field<T>* field)
     this->_degree = 0;
 }
 
+template <typename T>
+Polynomial<T>::~Polynomial()
+{
+    clear();
```

Comment:
Is it useful?

`clear` does two things:

- set `_degree` to 0: useless, the object is being destroyed
- calling `clear` on `terms`: IIRC `terms` is a map, so its destructor will be called here, there is no need to clear it beforehand, no?